extensions/wikia/AdEngine/js/video/player/porvata/floatingContextFactory.js

```diff
@@ -0,0 +1,103 @@
+/*global define, require*/
+define('ext.wikia.adEngine.video.player.porvata.floatingContextFactory', [
+		'ext.wikia.adEngine.video.player.porvata.floaterConfiguration',
+		'wikia.document',
+		'wikia.window'
+	], function (floaterConfiguration, doc, win) {
+		'use strict';
+
+		var events = {
+				attach: 'attach',
+				beforeDetach: 'beforeDetach',
+				detach: 'detach',
+				end: 'end',
+				start: 'start'
+			},
+			floatingThreshold = 55,
+			state = {
+				deactivated: 'deactivated',
+				floating: 'floating',
+				never: 'never',
+				paused: 'paused',
+				stopped: 'stopped'
+			};
+
+		function create(video, params, eventHandlers) {
+			var configuration = floaterConfiguration.selectConfigurationUsing(params),
+				adContainer = doc.getElementById(configuration.container),
+				elements = {
+					adContainer: adContainer,
+					ad: adContainer.querySelector('.wikia-ad'),
+					imageContainer: params.container.parentElement.querySelector('#image'),
```

Comment:
Is `#image` set by 3rd party? I wonder if there is a possibility we'll have more elements with `image` id which will make our page broken in terms of W3C standards.